src/commands/crashes/lib/symbols-uploading-helper.ts

```diff
@@ -0,0 +1,142 @@
+import { clientRequest, MobileCenterClient, models } from "../../../util/apis";
+import { ErrorCodes, failure } from "../../../util/commandline";
+import { DefaultApp } from "../../../util/profile";
+import { inspect } from "util";
+
+import * as JsZip from "jszip";
+import * as Crypto from "crypto";
+import * as _ from "lodash";
+import * as Request from "request";
+
+export default class SymbolsUploadingHelper {
+  constructor(private client: MobileCenterClient, private app: DefaultApp, private debug: Function) {}
+
+  public async uploadSymbolsZip(zip: JsZip): Promise<void> {
+    // get buffer for the prepared ZIP
+    const zipBuffer: Buffer = await this.getZipBuffer(zip);
+
+    // starting ZIP buffer MD5 hash calculation
+    const md5Hash: Promise<string> = this.calculateMd5(zipBuffer);
+
+    // executing API request to get an upload URL
+    const uploadingBeginRequestResult = await this.executeSymbolsUploadingBeginRequest(this.client, this.app);
+
+    // uploading
+    const symbolUploadId = uploadingBeginRequestResult.symbolUploadId;
+
+    try {
+      // doing HTTP PUT for ZIP buffer contents to the upload URL
+      const uploadUrl: string = uploadingBeginRequestResult.uploadUrl;
+      await this.uploadZipFile(uploadUrl, zipBuffer, await md5Hash);
+
+      // sending 'committed' API request to finish uploading
+      const uploadingEndRequestResult: models.SymbolUpload = await this.executeSymbolsUploadingEndRequest(this.client, this.app, symbolUploadId, "committed");
+    } catch (error) {
+      // uploading failed, aborting upload request
+      const uploadingAbortRequestResult = await this.abortUploadingRequest(this.client, this.app, symbolUploadId);
+      throw error;
+    }
+  }
+
+  private async getZipBuffer(zip: JsZip): Promise<Buffer> {
+    try {
+      this.debug("Getting in-memory ZIP archive as Buffer");
+      return await zip.generateAsync({
+        type: "nodebuffer"
+      });
+    } catch (error) {
+       throw failure(ErrorCodes.Exception, `Failed to compress the ZIP file: ${_.toString(error)}`);
+    }
+  }
+
+  private calculateMd5(buffer: Buffer): Promise<string> {
+    return new Promise<string>((resolve) => {
+      resolve(Crypto.createHash("md5").update(buffer).digest("base64"));
+    });
+  }
+
+  private async executeSymbolsUploadingBeginRequest(client: MobileCenterClient, app: DefaultApp): Promise<models.SymbolUploadBeginResponse> {
+    this.debug("Executing API request to get uploading URL");
+    let symbolUploadingBeginRequest = clientRequest<models.SymbolUploadBeginResponse>((cb) => client.symbols.postSymbolUpload(
```

Comment:
Why the separate variable and await call on line 69? Why not just await on this line here?